mudata/_core/mudata.py

```diff
@@ -349,31 +363,38 @@ def shape(self) -> Tuple[int, int]:
 
     def _update_attr(self, attr: str, axis: int, join_common: bool = False):
         """
-        Update global observations/variables with observations/variables for each modality
+        Update global observations/variables with observations/variables for each modality.
+
+        The following considerations are taken into account and will influence the time it takes to .update():
+        - are there duplicated obs_names/var_names?
+        - have obs_names/var_names of modalities changed?
         """
+
         prev_index = getattr(self, attr).index
 
-        self._check_duplicated_names()
+        # No _mod_index when upon read
+        # No _mod_index in mudata < 0.1.2
+        attr_names_maybe_changed = True
+        if hasattr(self, "_mod_index"):
+            for m, mod in self.mod.items():
+                if m in self._mod_index:
+                    if attr in self._mod_index[m]:
+                        attr_names_maybe_changed = attr_names_maybe_changed or not getattr(
```

Comment:
wont't `attr_names_maybe_changed` be always `True`, since you initialize it with `True` and then just `or` it? I think you want it to be `True` if there is no `_mod_index` attr, otherwise initialize it to `False` and `break` on the first unequal index